From 0871314d2fe64f1af4d647c510edcb63786ea5a2 Mon Sep 17 00:00:00 2001
From: Re van Dorst <opensource@vdorst.com>
Date: Fri, 12 Apr 2013 19:33:09 +0200
Subject: [PATCH 1/3] [LEDS] Changed remote trigger. Only triggers when key is
 known in table. LED keeps blinking when key is pressed.

---
 drivers/media/rc/rc-main.c |   16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/drivers/media/rc/rc-main.c b/drivers/media/rc/rc-main.c
index dfe2e1d..a74a931 100644
--- a/drivers/media/rc/rc-main.c
+++ b/drivers/media/rc/rc-main.c
@@ -26,7 +26,6 @@
 extern void ledtrig_rc_activity(void);
 #endif
 
-
 /* Sizes are in bytes, 256 bytes allows for 32 entries on x64 */
 #define IR_TAB_MIN_SIZE	256
 #define IR_TAB_MAX_SIZE	8192
@@ -611,6 +610,11 @@ void rc_repeat(struct rc_dev *dev)
 	if (!dev->keypressed)
 		goto out;
 
+#if defined(CONFIG_LEDS_TRIGGER_REMOTE_CONTROL)
+	/* LED Trigger Remote Control. */
+	ledtrig_rc_activity();
+#endif
+
 	dev->keyup_jiffies = jiffies + msecs_to_jiffies(IR_KEYPRESS_TIMEOUT);
 	mod_timer(&dev->timer_keyup, dev->keyup_jiffies);
 
@@ -652,14 +656,12 @@ static void ir_do_keydown(struct rc_dev *dev, int scancode,
 			   "key 0x%04x, scancode 0x%04x\n",
 			   dev->input_name, keycode, scancode);
 		input_report_key(dev->input_dev, keycode, 1);
-	}
-	input_sync(dev->input_dev);
-
 #if defined(CONFIG_LEDS_TRIGGER_REMOTE_CONTROL)
-	/* LED Trigger Remote Control. */
-	ledtrig_rc_activity();
+		/* LED Trigger Remote Control. */
+		ledtrig_rc_activity();
 #endif
-
+	}
+	input_sync(dev->input_dev);
 }
 
 /**
-- 
1.7.9.5

